src/lib/party.ts

```diff
@@ -0,0 +1,784 @@
+import { CircuitValue } from './circuit_value';
+import { Group, Field, Bool, VerificationKey } from '../snarky';
+import { PrivateKey, PublicKey } from './signature';
+import { UInt64, UInt32, Int64 } from './int';
+import * as Mina from './mina';
+
+export type Amount = UInt64;
+export const Amount = UInt64;
+export type Balance = UInt64;
+export const Balance = UInt64;
+export type Fee = UInt64;
+export const Fee = UInt64;
+export type GlobalSlot = UInt32;
+export const GlobalSlot = UInt32;
+export type SignedAmount = Int64;
+export const SignedAmount = Int64;
+
+const SnappStateLength: number = 8;
+
+/**
+ * Timing info inside an account.
+ */
+export class Timing {
+  initialMinimumBalance: Balance;
+  cliffTime: GlobalSlot;
+  cliffAmount: Amount;
+  vestingPeriod: GlobalSlot;
+  vestingIncrement: Amount;
+
+  constructor(
+    initialMinimumBalance: Balance,
+    cliffTime: GlobalSlot,
+    cliffAmount: Amount,
+    vestingPeriod: GlobalSlot,
+    vestingIncrement: Amount
+  ) {
+    this.initialMinimumBalance = initialMinimumBalance;
+    this.cliffTime = cliffTime;
+    this.cliffAmount = cliffAmount;
+    this.vestingPeriod = vestingPeriod;
+    this.vestingIncrement = vestingIncrement;
+  }
+}
```

Comment:
Classes like these, which don't have instance methods, should just be `type`s instead. That removes the need for a constructor and makes the code less painful. (This observation applies throughout this file)